### PR TITLE
Add product data context to Widget pages

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Add the product data context to widget pages (sign + preview)

## Justify why they are needed

- This allows us to use CMS blocks that depend on this context inside the widget!

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3628** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3628?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3632** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3632?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3633** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3633?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3635** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3635?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3635?utm_source=stack-comment).

**Review by guilhermespopolin — APPROVED** [review]
Awesome. Just a left a minor comment.

File: apps/store/src/pages/widget/preview/[[...slug]].tsx [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
   type: 'content' | 'flow'
-  [STORY_PROP_NAME]: PageStory | WidgetFlowStory
+  story: WidgetFlowStory
   hideChat?: boolean
+  productData: ProductData
 }
 
 const WidgetCmsPage = (props: PageProps) => {
   // PageStory and WidgetFlowStory are incompatible types but we don't actually care that
   // pros[STORY_PROP_NAME] gets properly typed here.
```
Comment (guilhermespopolin) [inline apps/store/src/pages/widget/preview/[[...slug]].tsx:29]: This can also be removed

**Review by guilhermespopolin — APPROVED** [review]

**Comment by robinandeer** [comment]
### Merge activity

* **Nov 24, 10:33 AM**: @robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3635).

**Pull request merged**